docs/eventTrigger.rst

```diff
@@ -0,0 +1,165 @@
+- **Trigger.schedule**: The schedule to perform this workflow on the regular basis. We use the cron time format here to specify, creating a trigger followed by the project workflow every 2 minutes 
+
+- **Trigger.schedule**: The params here is to clarify what regex pattern happening in the event coming from specific topic channel. The trigger kick-starts the flow if all of predefined dependency conditions are met. 
+
+Therefore, this trigger example will launch the flow once detecting Kafka event with anything in ``AzEvent_Topic4``, ``.*Partition[A-Z]....Event`` string in event comming from ``AzEvent_Topic4`` and ``hadoop?.*`` in ``AzEvent_Topic1``.
+
+The matching mechanism can be extended other than regex since now it is implemented as a generic interface.
+
+
+*****
+Event Based Trigger Example With Azkaban UI
+*****
+All scheduled data trigger will show up Azkaban Flow Trigger section. Also, project admins are able to pause and resume a scheduled trigger for undesirable situation.
+
+
+**Trigger info page for a specific flow:**
+
+.. image:: figures/TriggerInfo.png
+
+**Current and historic triggers for a specific flow:**
+
+.. image:: figures/TriggerList.png
+
+
+
+Follow these steps to run end to end local test:
+
+1.Start Kafka Broker Locally:
+
+Following `Kafka QuickStart <https://kafka.apache.org/quickstart/>`_ to run these Kafka console scripts in Kafka package
+::
+  *Start ZooKeeper
+  bin/zookeeper-server-start.sh config/zookeeper.properties
+  *Start Kafka Server
+  bin/kafka-server-start.sh config/server.properties
+
+2. Send Json event to topics with AzEvent_Topic4 as example:
+::
+  bin/kafka-console-producer.sh --broker-list localhost:9092 --topic AzEvent_Topic4 < recordPartition.json
+Here is how my ``recordPartition.json`` looks like:
+
+.. code-block:: json
+
+    {
+        "name":"Charlie", 
+        "team": "Azkaban",
+        "event":"MetastorePartitionAuditEvent"
+    }
+
+Once this event arrived, Azkaban will marked this specific event depedency as success. 
+
+.. image:: figures/Success.png
+
+3. Send another event from producer to launch the flow:
+::
+  bin/kafka-console-producer.sh --broker-list localhost:9092 --topic AzEvent_Topic4 < recordHadoop.json
+
+
+**Trigger the workflows that have all dependencies cleared out:**
+
+.. image:: figures/FlowExecuted.png
+
+*****
+Limitation
+*****
+Since our design purpose is to decouple the trigger condition from the action to take, currently there is a limitation on deserializing record. Although Kafka provides the ability to publish and subscribe to streams of records on custom serializer and deserializer. What we have right now is limited to Kafka build in String deserializer only. We are planning to enhance the flexibility on users to upload JAR with their own custom deserialize function in the near future. 
+
+*****
+Resource Referance
+*****
+- `Event Trigger Design Proposal <https://docs.google.com/document/d/1KxT17D8nvPvB-yTVZZEKuLMDp_TjdN1L6vagcZ4ypRI/edit>`_
```

Comment:
please remove these internal resources as this document is mostly for providing know-hows and instructions on the feature for OSS users.